sign.go

```diff
@@ -339,6 +348,48 @@ func (si *signerInfo) SetUnauthenticatedAttributes(extraUnsignedAttrs []Attribut
 	return nil
 }
 
+// TimestampTokenRequestCallback callback of timestamp token request.
+type TimestampTokenRequestCallback func(digest []byte) ([]byte, error)
+
+// RequestSignerTimestampToken add request of timestamp token with `signerID`
+// the request of timestamp token is called within `callback` function.
+func (sd *SignedData) RequestSignerTimestampToken(signerID int, callback TimestampTokenRequestCallback) error {
+	if len(sd.sd.SignerInfos) < (signerID + 1) {
+		return fmt.Errorf("no signer information found for ID %d", signerID)
+	}
+
+	if callback == nil {
+		return fmt.Errorf("no callback defined")
+	}
+
+	tst, err := callback(sd.sd.SignerInfos[signerID].EncryptedDigest)
+	if err != nil {
+		return err
+	}
+	return sd.AddTimestampTokenToSigner(signerID, tst)
+}
+
+// AddTimestampTokenToSigner inserts `tst` TimestampToken which described in RFC3161 into
+// unauthenticated attribute of `signerID`.
```

Comment:
It seems like the ID is an index? Might be useful to mention that? Or is this terminology used in this context?

Reply:
Updated at 12f70a22a9160d9d86634d72e04dcc7d023f1a9a